### PR TITLE
feat(internal): Add deprecated message warning

### LINKED ISSUE #994: Kraft v0.7.0 fails on Apple Silicon (Sonoma)
### Describe the bug

```
$ kraft fetch --log-type=basic
 W  specification in Kraftfile (v0.5) version is not latest (v0.6)
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x105182354]

goroutine 1 [running]:
kraftkit.sh/internal/cli/kraft/fetch.(*FetchOptions).pull(0x14000415c00, {0x1058affb8, 0x14000287350}, {0x1058c6058, 0x1400031bb00}, {0x16bc4f31d, 0x8}, 0x1, 0xffffffffffffffff)
        /__w/kraftkit/kraftkit/internal/cli/kraft/fetch/fetch.go:216 +0x674
kraftkit.sh/internal/cli/kraft/fetch.(*FetchOptions).Run(0x14000415c00, {0x1058affb8, 0x14000287350}, {0x0?, 0x0?, 0x0?})
        /__w/kraftkit/kraftkit/internal/cli/kraft/fetch/fetch.go:384 +0x17c
kraftkit.sh/cmdfactory.New.func1(0x0?, {0x140001a92a0?, 0x0?, 0x0?})
        /__w/kraftkit/kraftkit/cmdfactory/builder.go:365 +0x4c
kraftkit.sh/cmdfactory.AttributeFlags.bind.func3(0x14000235a00?, {0x140001a92a0, 0x1, 0x2})
        /__w/kraftkit/kraftkit/cmdfactory/builder.go:547 +0x150
github.com/spf13/cobra.(*Command).execute(0x14000160c00, {0x140001a9280, 0x2, 0x2})
        /go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:940 +0x658
github.com/spf13/cobra.(*Command).ExecuteC(0x14000160000)
        /go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:1068 +0x320
github.com/spf13/cobra.(*Command).Execute(...)
        /go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:992
github.com/spf13/cobra.(*Command).ExecuteContext(...)
        /go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:985
kraftkit.sh/cmdfactory.Main({0x1058affb8?, 0x140003a2f60}, 0x14000160000)
        /__w/kraftkit/kraftkit/cmdfactory/builder.go:183 +0x70
kraftkit.sh/internal/cli/kraft.Main({0x16bc4f300?, 0x1041eb6e0?, 0x10647d228?})
        /__w/kraftkit/kraftkit/internal/cli/kraft/kraft.go:165 +0x51c
main.main()
        /__w/kraftkit/kraftkit/cmd/kraft/main.go:19 +0x94
make: *** [unikraft] Error 2
```

### Steps to reproduce

Run `Kraft fetch`

### Expected behavior

_No response_

### Which architectures were you using or does this bug affect?

x86_64

### Which operating system were you using or does this bug affect?

macOS

### Relevant log output

_No response_

### LINKED ISSUE #1001: Add deprecated warning to hidden commands
### Feature request summary

Currently people who have existing scripts will not notice that a command became deprecated when upgrading kraftkit.

This will lead to confusion as to why it no longer works.

Adding a warning when running these commands should temporarily fix these problems.

### Describe alternatives

_No response_

### Related architectures

None

### Related platforms

None

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by craciunoiuc**
### Prerequisite checklist

  - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [x] Tested your changes against relevant architectures and platforms;
  - [x] Ran `make fmt` on your commit series before opening this PR;
  - [ ] Updated relevant documentation.

### Description of changes

GitHub-Fixes: #1001 
GitHub-Fixes: #994

### CODE REVIEW

**Comment by nderjung** [comment]
There is in fact a `Deprecated` attribute which can be set in the `cobra.Command` structure: https://github.com/spf13/cobra/blob/v1.8.0/command.go#L101-L102

**Comment by craciunoiuc** [comment]
Added this and now it looks like this fyi:

image

**Comment by craciunoiuc** [comment]
I liked the old format more, but I'm okay with this also

**Comment by nderjung** [comment]
I also prefer the old message since it works with our logger format. Is there a way to hi-jack that output and wrap it with our logger?

**Comment by craciunoiuc** [comment]
I don't think so, it does something like

`fmt.Printf("Command \"%s\" is deprecated, %s", command, msg)`

**Comment by nderjung** [comment]
Boo. Ok please revert back to your original work, this is fine. Thanks!

**Comment by craciunoiuc** [comment]
reverted

File: internal/cli/kraft/fetch/fetch.go [hunk 1]
```diff
@@ -56,10 +56,11 @@ func Fetch(ctx context.Context, opts *FetchOptions, args ...string) error {
 
 func NewCmd() *cobra.Command {
 	cmd, err := cmdfactory.New(&FetchOptions{}, cobra.Command{
-		Short:  "Fetch Unikraft unikernel dependencies",
-		Hidden: true,
-		Use:    "fetch [FLAGS] [DIR]",
-		Args:   cmdfactory.MaxDirArgs(1),
+		Short:      "Fetch Unikraft unikernel dependencies",
+		Hidden:     true,
+		Deprecated: "and should not be used",
```
Comment (nderjung) [inline internal/cli/kraft/fetch/fetch.go]: This can be removed no?

Comment (craciunoiuc) [reply internal/cli/kraft/fetch/fetch.go]: ye

**Review by nderjung — APPROVED** [review]

**Pull request merged**